examples/jsm/loaders/KTX2Loader.js

```diff
@@ -0,0 +1,537 @@
+
+			throw new Error( 'THREE.KTX2Loader: Only Basis Universal supercompression is currently supported.' );
+
+		}
+
+		if ( this.header.pixelDepth > 0 ) {
+
+			throw new Error( 'THREE.KTX2Loader: Only 2D textures are currently supported.' );
+
+		}
+
+		if ( this.header.arrayElementCount > 1 ) {
+
+			throw new Error( 'THREE.KTX2Loader: Array textures are not currently supported.' );
+
+		}
+
+		if ( this.header.faceCount > 1 ) {
+
+			throw new Error( 'THREE.KTX2Loader: Cube textures are not currently supported.' );
+
+		}
+
+
+		///////////////////////////////////////////////////
+		// Level index
+		///////////////////////////////////////////////////
+
+		var levelByteLength = this.header.levelCount * 3 * 8;
+		var levelReader = new KTX2BufferReader( this.arrayBuffer, idByteLength + headerByteLength, levelByteLength, littleEndian );
+
+		this.levels = [];
+
+		for ( var i = 0; i < this.header.levelCount; i ++ ) {
+
+			var level = {
+				byteOffset: levelReader.nextUint64(),
+				byteLength: levelReader.nextUint64(),
+				uncompressedByteLength: levelReader.nextUint64()
+			};
+
+			level.bytes = new Uint8Array( this.arrayBuffer, level.byteOffset, level.byteLength );
+
+			this.levels.push( level );
+
+		}
+
+
+		///////////////////////////////////////////////////
+		// Data Format Descriptor (DFD)
+		///////////////////////////////////////////////////
+
+		var dfdReader = new KTX2BufferReader(
+			this.arrayBuffer,
+			this.header.dfdByteOffset,
+			this.header.dfdByteLength,
+			littleEndian
+		);
+
+		this.dfd = {
+
+			vendorId: dfdReader.skip( 4 /* totalSize */ ).nextUint16(),
+			versionNumber: dfdReader.skip( 2 /* descriptorType */ ).nextUint16(),
+			colorModel: dfdReader.skip( 2 /* descriptorBlockSize */ ).nextUint8(),
+			colorPrimaries: dfdReader.nextUint8(),
+			transferFunction: dfdReader.nextUint8(),
+			flags: dfdReader.nextUint8(),
+
+			// ... remainder not implemented.
+
+		};
+
+
+		///////////////////////////////////////////////////
+		// Key/Value Data (KVD)
+		///////////////////////////////////////////////////
+
+		// Not implemented.
+		this.kvd = {};
+
+
+		///////////////////////////////////////////////////
+		// Supercompression Global Data (SGD)
+		///////////////////////////////////////////////////
+
+		var sgdReader = new KTX2BufferReader(
+			this.arrayBuffer,
+			this.header.sgdByteOffset,
+			this.header.sgdByteLength,
+			littleEndian
+		);
+
+		this.sgd = {
+
+			globalFlags: sgdReader.nextUint32(),
+			endpointCount: sgdReader.nextUint16(),
+			selectorCount: sgdReader.nextUint16(),
+			endpointsByteLength: sgdReader.nextUint32(),
+			selectorsByteLength: sgdReader.nextUint32(),
+			tablesByteLength: sgdReader.nextUint32(),
+			extendedByteLength: sgdReader.nextUint32(),
+
+			imageDescs: [],
+
+			endpointsData: null,
+			selectorsData: null,
+			tablesData: null,
+			extendedData: null,
+
+		};
+
+		for ( var j = 0; j < this.header.levelCount; j ++ ) {
```

Comment:
Nit: Any special reasons why you use `j`, not `i`, as loop var here?

Reply:
Oops, at the time I thought I'd hit one of those hoisting & block scope gotchas with `var` vs. `let`, but it was a mistake. This can be changed back to `i`.